index.test-d.ts

```diff
@@ -39,3 +39,17 @@ import Emittery = require('.');
 	const ee = new Emittery();
 	expectError(ee.on('anEvent', (data: any, more: any) => undefined));
 }
+
+// Strict typing
```

Comment:
There should probably be tests for `on`, `off`, `onAny`, `offAny` and really all public methods, which validate that the argument(s) received by listeners have the correct types. It looks like those tests didn't exist before though so conceivably could come in a different PR.